src/main/java/au/com/southsky/jfreesane/SaneDevice.java

```diff
@@ -214,4 +217,41 @@ SaneSession getSession() {
   void invalidateOptions() {
     optionTitleMap = null;
   }
+
+  @Override
+  public int hashCode() {
+    int hash = 3;
+    hash = 37 * hash + Objects.hashCode(this.name);
+    hash = 37 * hash + Objects.hashCode(this.vendor);
+    hash = 37 * hash + Objects.hashCode(this.model);
+    hash = 37 * hash + Objects.hashCode(this.type);
+    return hash;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
Maybe I overestimated my understanding of `jfreesane` and should have opened an issue rather than a PR in order to save us some time. My intention is to make SaneDevice serializable in order to change options and be able to use the changes directly after application restart (otherwise all changed options have to the tracked by the application). Imo the three classes should be sufficient to achieve that. Your points are valid and good. How to proceed? Should I open an issue with this comment as description or do you maybe want to take over this PR or do you just want to let me implement what you're commenting (might be inefficient)? Thanks for your feedback.